### PR TITLE
fix: format dates to WordPress `date_format` #3502

### LINKED ISSUE #3502: fix: format dates to WordPress `date_format`
## User Story

As a `Give Admin`, I want the `start/end date filter format` in the Donations admin page to match my locale so that `I have a consistent user experience`.

My website has UK English (en_GB) as the site language so I expect all dates to be formatted as `dd/mm/yyyy`.
## Current Behavior

Currently the date filter boxes have a placeholder of `mm/dd/yyyy` and when a date is selected from the dropdown calendar it is returned in the same US date format.

## Expected Behavior

I expect to see `dd/mm/yyyy` as the placeholder and have the date returned from the calendar dropdown formatted to `dd/mm/yyyy`.

## Acceptance Criteria

- [x] The date picker supports all date format (US or non-US) based on WordPress date settings.
- [x] Supporting WP date format does not affect other functionality like donation exports.

## Related
https://secure.helpscout.net/conversation/671557137/26233
https://secure.helpscout.net/conversation/658188284/25003?folderId=672197

### PR DESCRIPTION
**Pull request by jaydeeprami**
## Description
This PR will fix #3502

**Please note that:** WordPress Core has a open ticket regarding date format `jS F Y` so it will not work. Below is core ticket link: 
https://core.trac.wordpress.org/ticket/39991

## How Has This Been Tested?
Tested below cases:
- [X] Fix and Test all US and non-US format date setting which works fine in Datepicker
- [X] Fix and Test Export functionality for `dd/mm/yyyy` format and other format
- [X] Fix and Test Export functionality when `start_date` or `end_date` blank and export report

## Video (jpeg or gifs if applicable):
https://www.useloom.com/share/9973ff9b6785400b8c6a19efbe751066

## Types of changes
- Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style.
- [X] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/misc-functions.php [hunk 1]
```diff
@@ -2208,58 +2208,6 @@ function give_get_formatted_address( $address = array() ) {
...
-/**
- * Converts a PHP date format for use in JavaScript.
- *
- * @since 2.2.0
- *
- * @param string $php_format The PHP date format.
- *
- * @return string The JS date format.
- */
-function give_convert_php_date_format_to_js( $php_format ) {
```
Comment (ravinderk) [inline includes/misc-functions.php]: @jaydeeprami we cannot remove function like this add them to deprecate functions file if there is no use of them.

File: includes/admin/tools/export/give-export-donations-exporter.php [hunk 2]
```diff
@@ -278,10 +278,12 @@ public function get_donation_argument( $args = array() ) {
 		// Date query.
 		if ( ! empty( $this->start ) || ! empty( $this->end ) ) {
 			if ( ! empty( $this->start ) ) {
-				$defaults['date_query'][0]['after'] = date( 'Y-n-d 00:00:00', strtotime( $this->start ) );
+				$start_date                         = __give_get_mysql_format_date( $this->start );
+				$defaults['date_query'][0]['after'] = $start_date . ' 00:00:00';
```
Comment (ravinderk) [inline includes/admin/tools/export/give-export-donations-exporter.php]: @jaydeeprami use double quote

File: includes/admin/tools/export/give-export-donations-exporter.php [hunk 3]
```diff
@@ -278,10 +278,12 @@ public function get_donation_argument( $args = array() ) {
...
 			}
 			if ( ! empty( $this->end ) ) {
-				$defaults['date_query'][0]['before'] = date( 'Y-n-d 00:00:00', strtotime( $this->end ) );
+				$end_date                            = __give_get_mysql_format_date( $this->end );
+				$defaults['date_query'][0]['before'] = $end_date . ' 23:59:59';
```
Comment (ravinderk) [inline includes/admin/tools/export/give-export-donations-exporter.php]: @jaydeeprami use double quote

**Comment by jaydeeprami** [comment]
@ravinderk 

I have made all change as you suggested.

**Pull request merged**